### DISCUSSION TITLE
How to install traefik on agent nodes?

### DISCUSSION THREAD

**Discussion by HouzuoGuo** [k3s-io/k3s#9972] [Q&A] [answered]
Hey. I've got a cluster of 5 nodes comprising 2 servers and 3 agents. This page explains that traefik is automatically installed for servers alone:
https://docs.k3s.io/networking/networking-services#traefik-ingress-controller

The agents have less memory and cannot join as servers, but I would like them to run traefik just as servers do. Is there a way to do that with k3s?

Thanks!

**Comment by brandond** [comment, marked as answer]
> This page explains that traefik is automatically installed for servers alone

It does not say that anywhere, nor is that true. Traefik runs only a single replica by default, so it will run on whatever node it can first be scheduled to. If you want more replicas, you can use `kubectl scale deployment` to ask for additional pods to be run.

Note that by default the ServiceLB daemonset allows access to the traefik pod from all nodes, even if there is only a single replica running on a single node. This is also covered in the docs:
> The Traefik ingress controller deploys a LoadBalancer Service that uses ports 80 and 443. By default, ServiceLB will expose these ports on all cluster members.
